apps/extension/src/Approvals/ApproveTransfer/ConfirmLedgerTransfer.tsx

```diff
@@ -0,0 +1,103 @@
+import { useCallback, useState } from "react";
+import { LedgerError } from "@anoma/ledger-namada";
+import { toBase64 } from "@cosmjs/encoding";
+
+import { Button, ButtonVariant } from "@anoma/components";
+
+import { Ledger } from "background/ledger";
+import {
+  GetTransferBytesMsg,
+  SubmitSignedTransferMsg,
+} from "background/ledger/messages";
+import { Ports } from "router";
+import { closeCurrentTab } from "utils";
+import { useRequester } from "hooks/useRequester";
+import { Status } from "Approvals/Approvals";
+import {
+  ApprovalContainer,
+  ButtonContainer,
+} from "Approvals/Approvals.components";
+
+type Props = {
+  msgId: string;
+};
+
+export const ConfirmLedgerTransfer: React.FC<Props> = ({ msgId }) => {
+  const requester = useRequester();
+  const [error, setError] = useState<string>();
+  const [status, setStatus] = useState<Status>();
+
+  const submitTransfer = async (): Promise<void> => {
+    setStatus(Status.Pending);
+    const ledger = await Ledger.init();
+
+    try {
+      // Constuct tx bytes from SDK
+      const { bytes, path } = await requester.sendMessage(
+        Ports.Background,
+        new GetTransferBytesMsg(msgId)
+      );
+
+      // Sign with Ledger
+      const signatures = await ledger.sign(bytes, path);
+      const { errorMessage, returnCode } = signatures;
+
+      // Close transport so that it may be re-opened on a subsequent attempt (due to error)
+      await ledger.closeTransport();
+
+      if (returnCode !== LedgerError.NoErrors) {
+        setError(errorMessage);
+        return setStatus(Status.Failed);
+      }
+
+      // Submit signatures for tx
+      await requester.sendMessage(
+        Ports.Background,
+        new SubmitSignedTransferMsg(msgId, toBase64(bytes), signatures)
+      );
+      setStatus(Status.Completed);
+    } catch (e) {
+      console.warn(e);
+      const ledgerErrors = await ledger.queryErrors();
```

Comment:
The only error that can happen is that the Ledger instance can't find `NamadaApp` - so either the app is not loaded, or the Ledger isn't plugged in an connected. The query for `status()` in `background/ledger/ledger.ts` catches that situation there, this is more of a wrapper function for `status()` that returns any non-error error messages so the client doesn't have to do that logic. However, if that error was thrown in the `Ledger` status call, it isn't getting caught here, which isn't good. Maybe it should be moved to the above `try` so that the status error, if thrown, will be caught.

Reply:
Also, I just realized that this is `ConfirmLedgerTransfer` - in my next PR I am getting rid of this file entirely - I took out the `queryErrors` call in `ConfirmLedgerBond`, so I'll leave this as is! Sorry I thought this was the other confirmation!